src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts

```diff
@@ -1,75 +1,209 @@
+          break;
+        case 'float':
+        case 'unfilterable-float':
+          fragColorType = 'f32';
+          suffix = '';
+          fractionDigits = 4;
+          break;
+        default:
+          unreachable();
+      }
 
-    const v = output.map(n => n.toFixed(fractionDigits));
+      let outputType;
+      const v = o.values.map(n => n.toFixed(fractionDigits));
+      switch (o.componentCount) {
+        case 1:
+          outputType = fragColorType;
+          resultStrings.push(`${v[0]}${suffix}`);
+          break;
+        case 2:
+          outputType = `vec2<${fragColorType}>`;
+          resultStrings.push(`${outputType}(${v[0]}${suffix}, ${v[1]}${suffix})`);
+          break;
+        case 3:
+          outputType = `vec3<${fragColorType}>`;
+          resultStrings.push(`${outputType}(${v[0]}${suffix}, ${v[1]}${suffix}, ${v[2]}${suffix})`);
+          break;
+        case 4:
+          outputType = `vec4<${fragColorType}>`;
+          resultStrings.push(
+            `${outputType}(${v[0]}${suffix}, ${v[1]}${suffix}, ${v[2]}${suffix}, ${v[3]}${suffix})`
+          );
+          break;
+        default:
+          unreachable();
+      }
 
-    let outputType;
-    let result;
-    switch (componentCount) {
-      case 1:
-        outputType = fragColorType;
-        result = `${v[0]}${suffix}`;
-        break;
-      case 2:
-        outputType = `vec2<${fragColorType}>`;
-        result = `${outputType}(${v[0]}${suffix}, ${v[1]}${suffix})`;
-        break;
-      case 3:
-        outputType = `vec3<${fragColorType}>`;
-        result = `${outputType}(${v[0]}${suffix}, ${v[1]}${suffix}, ${v[2]}${suffix})`;
-        break;
-      case 4:
-        outputType = `vec4<${fragColorType}>`;
-        result = `${outputType}(${v[0]}${suffix}, ${v[1]}${suffix}, ${v[2]}${suffix}, ${v[3]}${suffix})`;
-        break;
-      default:
-        unreachable();
+      outputStructString += `@location(${i}) o${i} : ${outputType},\n`;
     }
 
     return `
-    @stage(fragment) fn main() -> @location(0) ${outputType} {
-        return ${result};
+    struct Outputs {
+      ${outputStructString}
+    }
+
+    @stage(fragment) fn main() -> Outputs {
+        return Outputs(${resultStrings.join(',')});
     }`;
   }
 }
 
 export const g = makeTestGroup(F);
 
+g.test('color,attachments')
+  .desc(`Test that pipeline with  sparse color attachments write values correctly.`)
+  .params(u =>
+    u
+      .combine('format', kRenderableColorTextureFormats)
+      .beginSubcases()
+      .combine('attachmentCount', [2, 3, 4])
+      .expand('emptyAttachmentId', p => range(p.attachmentCount, i => i))
+  )
+  .fn(async t => {
+    const { format, attachmentCount, emptyAttachmentId } = t.params;
+    const componentCount = kTexelRepresentationInfo[format].componentOrder.length;
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    const writeValue = { R: 1, G: 1, B: 1, A: 1 };
```

Comment:
Probably want a different writeValue for each attachment so we can make sure they didn't get mixed up somehow

Reply:
Fixed (also set different value for sample type = float/int)